horace_core/sqw/@TmpFileHandler/TmpFileHandler.m

```diff
@@ -38,12 +43,28 @@
             end
         end
         function is = isempty(obj)
-            % for checking this class holder property for it beeing 
+            % for checking this class holder property for it beeing
             % empty (not assigned anything) or invalid (delete method
             % called directrly)
             is = ~isvalid(obj);
         end
+        function obj=copy(obj)
+            obj.copy_count_ = obj.copy_count_+1;
+        end
+        function lock(obj)
+            obj.copy_count_ = obj.copy_count_-1;
+            if obj.copy_count_ < 1
+                obj.is_locked_ = true;
+            end
+        end
+        function unlock(obj)
+            obj.is_locked_ = false;
+        end
+
```

Comment:
Some tests are failing otherwise. Good enough reason for me. May be can be done better (you may notice I am adding and removing this functionality as new issue occur) but I have not currently came with better way of doing thins. 

working non-ideally now is better then not working now and woring ideally in a future.